### PR TITLE
DBZ-784 Terminate execute loops during shutdown

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-784

### CODE REVIEW

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java [hunk 1]
```diff
@@ -386,6 +397,17 @@ public void executeBlocking(String desc, BlockingConsumer<MongoClient> operation
                 return collections;
             });
         }
+
+        private boolean isRunning() {
```
Comment (jchipmunk) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java]: I see the discrepancy between the method name and the variable name. The method returns `true` if `MongoPrimary` is stopped.

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java [hunk 2]
```diff
@@ -272,6 +274,9 @@ public void execute(String desc, Consumer<MongoClient> operation) {
                     return;
                 } catch (Throwable t) {
                     errorHandler.accept(desc, t);
+                    if (!isRunning()) {
```
Comment (jchipmunk) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java]: If `stop` variable has `false` value (it is default value for `AtomicBoolean`), then `ConnectException` is thrown here, and `errorMetronome.pause()` will never be called.

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java]: @jchipmunk Sigh, you are right, the correct code meant to be
`return !stop.get()`

Sorry for that :-(

Comment (jpechane) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ConnectionContext.java]: @jchipmunk Please check https://github.com/debezium/debezium/pull/636

**Pull request merged**